src/spikeinterface/sortingcomponents/tests/test_wobble.py

```diff
@@ -0,0 +1,52 @@
+import pytest
+import numpy as np
+
+from spikeinterface.sortingcomponents.matching import wobble
+
+
+def test_compute_objective():
+    # Arrange: Generate random 'data'
+    num_templates = np.random.randint(1, 100)
+    num_samples = np.random.randint(10, 20)
+    conv_approx_rank = np.random.randint(1, num_samples)
+    num_channels = np.random.randint(1, 100)
+    chunk_len = np.random.randint(num_samples*2, num_samples*10)
+    traces = np.random.rand(chunk_len, num_channels)
+    temporal = np.random.rand(num_templates, num_samples, conv_approx_rank)
+    singular = np.random.rand(num_templates, conv_approx_rank)
+    spatial = np.random.rand(num_templates, conv_approx_rank, num_channels)
+    compressed_templates = (temporal, singular, spatial, temporal)
+    norm = np.random.rand(num_templates)
+    template_data = wobble.TemplateData(compressed_templates=compressed_templates, pairwise_convolution=[], norm=norm)
+
+    # Act: run compute_objective
+    objective, objective_normalized = wobble.compute_objective(traces, template_data, conv_approx_rank)
+
+    # Assert: check shape
+    assert objective.shape == (num_templates, chunk_len+num_samples-1)
```

Comment:
What I was proposing before is for you to use copy the `compute_objective` method that you have now written as loops (it is just simple and readable). Then, when you start with newer implementations that vectorize the loops or do more fnacy things then will have to agree this function here. This will both serve as a test and documentation as usuallly the code becomes less clear once you start vectorizing.

Reply:
Oh, I see.  Thanks for the clarification, I've added that now.